Dockerfile

```diff
@@ -1,5 +1,9 @@
-FROM rust AS builder
+FROM ubuntu:latest AS git-deps
+RUN apt update && apt install -yq git
+COPY . /tmp/
```

Comment:
Can you simplify this to enable better caching?

Write now this build phase is run each time any file changes.
You see how the rust builder only re-runs when files in operator change?
`COPY .` is a big pull. Maybe there is much less info needed to get the submodules, so it only re-runs when the remote references are run. Can you check?

Reply:
yes, resolved with 83d18f93430801637c5dc9fd448ef8e5ab9fb41f

Only submodule updates if the .gitmodules or contracts/lib directory is changed.